Webpack DLL

kane-c opened the pull request:
This massively speeds up recompiling and hot reloading.

~~The only downside is the DLL must be built before starting the webpack dev server for the first time, and again after updating installed modules.~~

~~Use `npm run build-dll` to build it.~~

The initial list of modules included in the DLL is based on the largest modules in the browser bundle according to the output of the Webpack bundle analysis. Feel free to adjust it.

kane-c committed:
Remove outdated comment

kane-c committed:
Add Webpack DLL during development

This massively speeds up recompiling and hot reloading.

The only downside is the DLL must be built before starting
the webpack dev server for the first time, and again after
updating installed modules.

The initial list of modules included in the DLL is based on
the largest modules in the browser bundle according to
the output of the Webpack bundle analysis.

leebenson commented:
Thanks for submitting this PR @kane-c. Supporting DLL builds is great, but it's too big a breaking change to require `npm run build-dll` before running `npm start`. If we're going to add this, it needs to be done seamlessly / as part of the same build step.

Also, imports that reference the `dist` folder such as [this one](https://github.com/reactql/kit/pull/108/files#diff-7f83d38359798eccf58076973b0d06beR26) folder are inherently brittle - the folder is intended to be ephemeral and not a constant part of the build system; only files that are already in `dist` should (locally) reference siblings files.

If the above can be considered, I'm all for adding this.

kane-c commented:
Thanks for the feedback, couldn't agree more.

I've moved the DLL manifest into `kit/webpack` (let me know if there's a better place for it) instead of in `dist`.

Also I added the DLL build step into the `browser` npm script - the impact on the initial build time is a lot smaller than I thought so it's harmless there. This way there is no breaking change :)

leebenson commented:
Closing due to releasing a new [3.0.0 kit](https://github.com/leebenson/reactql/releases/tag/3.0.0) - please reopen at https://github.com/leebenson/reactql if you feel it's still relevant, thanks.
